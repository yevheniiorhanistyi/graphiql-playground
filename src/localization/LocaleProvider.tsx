import { FC, ReactNode, useEffect, useMemo, useState } from 'react';
import { LOCAL_STORAGE_LOCALE_KEY, Locale, LocaleContext } from './LocaleContext';
import { getFromLocalStorage } from '@/utils/localStorageService';

const LocaleProvider: FC<{ children: ReactNode }> = ({ children }) => {
  const [locale, setLocale] = useState<Locale>(Locale.EN);

  useEffect(() => {
    const storedLocale = getFromLocalStorage(LOCAL_STORAGE_LOCALE_KEY) as Locale;
    setLocale(storedLocale || Locale.EN);
  }, []);

  const defaultProps = useMemo(() => ({ locale, setLocale }), [locale]);

  return <LocaleContext.Provider value={defaultProps}>{children}</LocaleContext.Provider>;
};

export default LocaleProvider;
