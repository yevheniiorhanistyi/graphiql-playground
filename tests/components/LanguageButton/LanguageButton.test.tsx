import { vi, describe, test, expect } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';

import LocaleProvider from '@/localization/LocaleProvider';
import LangButton from '@/components/LanguageButton/LanguageButton';

const handleOpenMock = vi.fn();
const handleCloseMock = vi.fn();

describe('LangButton', () => {
  test('renders LangButton correctly', () => {
    const { container } = render(
      <LangButton isPopoverOpen={false} handleOpen={handleOpenMock} handleClose={handleCloseMock} />
    );
    expect(container).toMatchSnapshot();
  });

  test('calls handleOpen when button is clicked', () => {
    render(
      <LocaleProvider>
        <LangButton
          isPopoverOpen={false}
          handleOpen={handleOpenMock}
          handleClose={handleCloseMock}
        />
      </LocaleProvider>
    );

    const langButton = screen.getByTestId('language-button');

    fireEvent.click(langButton);

    expect(handleOpenMock).toHaveBeenCalled();
  });

  test('calls onItemClick and handleClose when an item English is clicked', () => {
    render(
      <LocaleProvider>
        <LangButton
          isPopoverOpen={false}
          handleOpen={handleOpenMock}
          handleClose={handleCloseMock}
        />
      </LocaleProvider>
    );

    const langItem = screen.getByText('English');

    fireEvent.click(langItem);

    expect(handleCloseMock).toHaveBeenCalled();
  });

  test('calls onItemClick and handleClose when an item Russian is clicked', () => {
    render(
      <LocaleProvider>
        <LangButton
          isPopoverOpen={false}
          handleOpen={handleOpenMock}
          handleClose={handleCloseMock}
        />
      </LocaleProvider>
    );

    const langItem = screen.getByText('Русский');

    fireEvent.click(langItem);

    expect(handleCloseMock).toHaveBeenCalled();
  });
});
