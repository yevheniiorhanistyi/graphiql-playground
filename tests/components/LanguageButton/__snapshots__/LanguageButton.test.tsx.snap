// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LangButton > renders LangButton correctly 1`] = `
<div>
  <button
    class="_langBtn_f843e0"
    data-testid="language-button"
    type="button"
  >
    <ul
      class="_langPopover_f843e0"
    >
      <li
        class="_langPopover_item_f843e0"
      >
        <img
          alt="English"
          class="_flagIcon_f843e0"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/en_flag.svg"
          style="color: transparent;"
          width="28"
        />
        <span
          class="_languageText_f843e0"
        >
          English
        </span>
      </li>
      <li
        class="_langPopover_item_f843e0"
      >
        <img
          alt="Русский"
          class="_flagIcon_f843e0"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/ru_flag.svg"
          style="color: transparent;"
          width="28"
        />
        <span
          class="_languageText_f843e0"
        >
          Русский
        </span>
      </li>
    </ul>
  </button>
</div>
`;
